### DISCUSSION TITLE
antv/x6-react-shape  使用react组件  使用useContext 永远是默认值  如何使用状态管理

### DISCUSSION THREAD

**Discussion by meikang123** [antvis/X6#1844] [Q&A]
antv/x6-react-shape  使用react组件  使用useContext 永远是默认值  如何使用状态管理

**Comment by NewByVector** [comment]
因为 react 节点不处于正常的 reactdom 树中，所以状态是默认的，可以使用这种方式解决：https://github.com/antvis/X6/blob/master/examples/x6-example-features/src/pages/react/portal.tsx
